lib/ddtrace/distributed_tracing/headers/datadog.rb

```diff
@@ -28,7 +28,7 @@ def self.extract(env)
           # Return early if this propagation is not valid
           # DEV: To be valid we need to have a trace id and a parent id or when it is a synthetics trace, just the trace id
           # DEV: `DistributedHeaders#id` will not return 0
-          return unless (trace_id && parent_id) || (origin == 'synthetics' && trace_id)
+          return unless (trace_id && parent_id) || (origin && origin.include?('synthetics') && trace_id)
```

Comment:
You can change this to be `(origin && trace_id)`

Lets not worry about the value of the origin header. Basically, if an origin is provided, then `X-Datadog-Parent-Id` is optional.